### ISSUE TITLE
Some exclusion mechanism for NullableToStringCall

### ISSUE THREAD

**Issue by vladimirfx** [detekt/detekt#3855]
## Expected Behavior of the rule

It will be great to have some method definition/call side exclusion mechanism for NullableToStringCall.

## Context

NullableToStringCall is very useful for us and with its help, we detect a couple of real bugs.

But >90% of reports are useless because it reports the code where a nullable String is ok and intended:
- all kinds of log/trace messages
- toString() implementations itself

Tipical toString() impl:

``` kotlin
    override fun toString(): String =
        "Notification(" +
            "clientId=$clientId, " +
            "message=$message, " +
            "type=$type, " +
            "style=$style, " +
            "createdDate=$createdDate, " +
            "relevanceDate=$relevanceDate, " +
            "showedDate=$showedDate, " +
            ")"
```

Some of the class fields (obviously) can be nullable. As a result, NullableToStringCall is the most frequently detected issue in our codebase regardless of the project. So we are forced to disable this useful rule in most projects because of the very high maintenance cost of suspending reports case by case.

**Comment by cortinico** [comment]
Can you be a bit more specific on what would be a valid exclusion pattern?
Other than the mentioned example, are there other scenarios where a nullable string is accepted?

**Comment by vladimirfx** [comment]
1. Most unnecessary flagged code becomes to `toString()` overrides or properties of String type. So it is function/property definition side. There is exclusion by contained function/property id needed.

2. Other cases come to logging calls. 

And what is interesting, this code NOT flagged:
``` kotlin
log.info("This is ${this::class.simpleName}.")
```

But this is flagged:

``` kotlin
val msg = "This is ${this::class.simpleName}."
log.info(msg)
```

80% of our cases can be covered by excluding `toString()` overrides (even with hardcode in rule).

**Comment by BraisGabin** [comment]
Maybe a `ignoreToString` could be implemented. Then we need to work on a more powerful way of ignore issues. It was planned for 2.0 but I'm not sure anymore... There is a lot of users asking for it right now.

> And what is interesting, this code NOT flagged:
> 
> ```kotlin
> log.info("This is ${this::class.simpleName}.")
> ```
> 
> But this is flagged:
> 
> ```kotlin
> val msg = "This is ${this::class.simpleName}."
> log.info(msg)
> ```

🤔 That sounds like a false negative to me. What does `log.info()` want as parameter? Could you open another issue so we can look to it?

**Comment by vladimirfx** [comment]
> 🤔 That sounds like a false negative to me. What does `log.info()` want as parameter? Could you open another issue so we can look to it?

https://github.com/apache/logging-log4j-kotlin/blob/87a6a017f06498c2ba49044c89ad639eda3f5d6c/log4j-api-kotlin/src/main/kotlin/org/apache/logging/log4j/kotlin/KotlinLogger.kt#L339

Not this simple CharSequence variant nor Supplier based (lazy) variant raise an issue.

If this will be fixed we will get event more not relevant reports from NullableToStringCall. So exclusion mechanism becomes a must-have for us. But in this case, exclusion should work on the call side and I have no idea how it can be described in config...
